src/Arcus.WebApi.Logging.Core/Correlation/HttpCorrelationClientOptions.cs

```diff
@@ -56,5 +57,24 @@ public string TransactionIdHeaderName
                 _transactionIdHeaderName = value;
             }
         }
+
+        /// <summary>
+        /// Gets the telemetry context used during HTTP dependency tracking.
+        /// </summary>
+        internal Dictionary<string, object> TelemetryContext { get; private set; } = new Dictionary<string, object>();
+
+        /// <summary>
+        /// Adds a telemetry context while tracking the HTTP dependency.
+        /// </summary>
+        /// <param name="telemetryContext">The dictionary with contextual information about the dependency telemetry.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="telemetryContext"/> is <c>null</c>.</exception>
+        public void AddTelemetryContext(Dictionary<string, object> telemetryContext)
+        {
+            Guard.NotNull(telemetryContext, nameof(telemetryContext), "Requires a telemetry context dictionary to add to the HTTP dependency tracking");
+            foreach (KeyValuePair<string, object> item in telemetryContext)
+            {
+                TelemetryContext.Add(item.Key, item.Value);
```

Comment:
What if there already exists an entry for `item.Key` ?  I think we should check on that, and decide if we overwrite or not (I'd say overwrite)

Reply:
Ah, then we could just use `dictionary[key] = value` for that, as that is the default behavior. The `Add` will throw.

Reply:
Done in https://github.com/arcus-azure/arcus.webapi/pull/371/commits/0780250730751c017a507e32622d157f263c0d8b